### ISSUE TITLE
Support for Torznab API

### ISSUE THREAD

**Issue by tylergets** [sergiotapia/magnetissimo#56]
It would be awesome to the support the Torznab api, as it would allow many automation tools to work with this project. 

Here is documentation from Sonarr, an automation tool.
https://github.com/Sonarr/Sonarr/wiki/Implementing-a-Torznab-indexer

**Comment by sergiotapia** [comment]
I'd love to implement this. Thanks for the heads up!

**Comment by agustif** [comment]
http://torrent2magnet.com/

https://github.com/ungoldman/magnet-link

**Comment by lukyjay** [comment]
It has been 11 months, has there been any progress on this?

**Comment by tchoutri** [comment]
@lukyjay no. The short-term goal is to ensure that all of the existing crawlers work properly, then I will take a look at that. I'll ping you when it's implemented.

Happy Holidays.

**Comment by tchoutri** [comment]
@lukyjay Implementing a Torznab API crawler would go against our "Run without ceremony" goal. This is a self-hosted service so each user would have to enter an endpoint URL and perhaps API credentials.
We currently do not have a mechanism to start/stop crawlers at runtime, so we cannot include a torznab crawler that wouldn't be bothering users who don't use it.
I'm closing this issue for now, we'll open it when it becomes technically possible to provide a Torznab crawler.

**Comment by gjedeer** [comment]
Hey @tchoutri of course as a contributor you have the deciding voice here, but I think you're misunderstanding what this ticket is about. Implementing a Torznab-compatible API means that a GET request to

    /api?t=search&cat=..,..&q=Anime+Title+123

would return an XML with search results in the already harvested magnets. It's just a different format for requesting searches, not a way to fire crawlers.

**Comment by sergiotapia** [comment]
Right, this we can do most definitely. It's just spitting out Torznad formatted XML.

**Comment by tchoutri** [comment]
Oh then it's not a problem at all. :)

**Comment by sergiotapia** [comment]
It's been a ludicrous amount of time but I'm going to tackle this tomorrow. 👿

**Comment by askiiart** [comment]
@sergiotapia Was this finally added? If so, how can I add it to *arr? I'm just getting an error when I try to connect  (I'm using `http://x.x.x.x:4000/api`)

**Comment by lukyjay** [comment]
> @sergiotapia Was this finally added? If so, how can I add it to *arr? I'm just getting an error when I try to connect  (I'm using `http://x.x.x.x:4000/api`)

He said he'd make a start, not finish 🙂

**Comment by ldexterldesign** [comment]
👋

Has this task been [done]?: if yes then this issue can be closed

Hope this helps

Regards

[done]: https://github.com/sergiotapia/magnetissimo/issues/145

**Comment by askiiart** [comment]
@ldexterldesign I believe issue #145 was closed because it's a duplicate of this.

**Comment by ldexterldesign** [comment]
BUMP

If this issue gets done then can setup prowlarr, integrate with the other *arr apps and start making use of this app 🙂

Then I would do https://github.com/sergiotapia/magnetissimo/issues/162

Regards

**Comment by ldexterldesign** [comment]
👋 @sergiotapia,

Hope you're OK!

Thanks for continued work on this

Did you mean to create a duplicate [issue]?

Is sonarr et al. supported yet..?: if yes then I'll update and have a play 👀

Yours hopefully

[issue]: https://github.com/sergiotapia/magnetissimo/issues/171
